services/order.py

```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> None:
+    with transaction.atomic():
+        user_id = get_user_model().objects.get(username=username).id
+        order = Order.objects.create(user_id=user_id)
+        if date:
+            order.created_at = date
+            order.save()
+
+    for ticket_data in tickets:
+        Ticket.objects.create(
+            row=ticket_data["row"],
+            seat=ticket_data["seat"],
+            movie_session_id=ticket_data["movie_session"],
+            order=order
+        )

```

Comment:
so why do u use transaction here, if tickets creation is out of transaction scope ?